tests/tl/test_differential_expression_analysis.py

```diff
@@ -0,0 +1,412 @@
+from unittest.mock import patch
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from alphastats.dataset.keys import Cols
+from alphastats.dataset.preprocessing import PreprocessingStateKeys
+from alphastats.tl.differential_expression_analysis import (
+    DeaColumns,
+    DeaParameters,
+    DifferentialExpressionAnalysis,
+    DifferentialExpressionAnalysisTTest,
+    DifferentialExpressionAnalysisTwoGroups,
+)
+
+
+class TestableDifferentialExpressionAnalysis(DifferentialExpressionAnalysis):
+    @staticmethod
+    def _allowed_parameters():
+        return [DeaParameters.METADATA]
+
+    def _extend_validation(self, parameters):
+        if parameters[DeaParameters.METADATA] is None:
+            raise ValueError("Metadata must be provided")
+
+    def _run_statistical_test(self, **kwargs):
+        return self._statistical_test_fun(
+            self.input_data, metadata=kwargs[DeaParameters.METADATA]
+        )
+
+    @staticmethod
+    def _statistical_test_fun(input_data, metadata):
+        return pd.DataFrame(
+            [[0.055, 2, 0.009]],
+            columns=[DeaColumns.PVALUE, DeaColumns.LOG2FC, DeaColumns.QVALUE],
+            index=["gene1"],
+        )
+
+
+def test_dea_no_abstractmethods():
+    with pytest.raises(TypeError):
+        DifferentialExpressionAnalysis(None)
+
+
+def test_dea_perform_success():
```

Comment:
docstrings would help here, also using `match` for the individual `Errors` would increase readability and robustness (there are multiple different `Value` and `KeyErrors`)